lua/wikis/commons/Infobox/Person.lua

```diff
@@ -242,6 +226,72 @@ function Person:createInfobox()
 	return self:build(widgets)
 end
 
+function Person:_parseArgs()
+	local args = self.args
+
+	-- STATUS and BANNED
+	local lowerStatus = (args.status or ''):lower()
+	if lowerStatus == BANNED then
+		-- Temporary until conversion
+		args.banned = args.banned or true
+	end
+	args.status = STATUS_TRANSLATE[lowerStatus]
+
+	-- ENRICH TEAM
+	if Logic.readBool(args.autoTeam) then
+		local team, team2 = PlayerIntroduction.playerTeamAuto{player=self.pagename}
+		args.team = Logic.emptyOr(args.team, team)
+		args.team2 = Logic.emptyOr(args.team2, team2)
+	end
+
+	-- COUNTRY/REGION
+
+	self.locations = self:getLocations()
+	-- check if non-representing is used and set an according value in self
+	args.country = self:getStandardNationalityValue(args.country or args.nationality)
+	if args.country == self:getStandardNationalityValue('non-representing') then
+		self.nonRepresenting = true
+	end
+	self.region = Region.run({region = args.region, country = args.country})
+
+	-- NAME
+	args.ids = args.ids or args.alternateids
+	args.givenname, args.familyname = self:_flipNameOrder(args)
+
+	-- EARNINGS
+	self.totalEarnings, self.earningsPerYear = self:calculateEarnings(args)
+
+	-- AGE
+	local ageCalculationSuccess, age = pcall(AgeCalculation.run, {
+			birthdate = args.birth_date,
+			birthlocation = args.birth_location,
+			deathdate = args.death_date,
+			deathlocation = args.death_location,
+		})
+	if not ageCalculationSuccess then
+		age = self:_createAgeCalculationErrorMessage(age --[[@as string]])
+	end
+
+	self.age = age
+
+	-- ROLES
+	-- Backwards compatibility for the old roles
+	if not args.roles then
+		args.roles = table.concat({
+			args.role,
+			args.role2,
+			args.role3,
+		}, ', ')
```

Comment:
could we have edge cases with role3 set and role2 unset? (or role2 set and role1 unset)
if so we could get errors here